doc/authentication.md

```diff
@@ -0,0 +1,162 @@
+1. Place the certificates issued by the CA on the clients and the server, replacing the automatically generated ones.
+1. Restart the server.
+
+In that mode, any connection to a LXD daemon will be done using the
+preseeded CA certificate.
+
+If the server certificate isn't signed by the CA, the connection will simply go through the normal authentication mechanism.
+If the server certificate is valid and signed by the CA, then the connection continues without prompting the user for the certificate.
+
+Note that the generated certificates are not automatically trusted. You must still add them to the server in one of the ways described in {ref}`authentication-trusted-clients`.
+
+(authentication-candid)=
+## Candid-based authentication
+
+When LXD is configured to use [Candid](https://github.com/canonical/candid) authentication, clients that try to authenticate with the server must get a Discharge token from the authentication server specified by the `candid.api.url` setting (see {doc}`server`).
+
+The authentication server certificate must be trusted by the LXD server.
+
+To add a remote pointing to a LXD server configured with Candid/Macaroon authentication, run `lxc remote add REMOTE ENDPOINT --auth-type=candid`.
+To verify the user, the client will prompt for the credentials required by the authentication server.
+If the authentication is successful, the client will connect to the LXD server and present the token received from the authentication server.
+The LXD server verifies the token, thus authenticating the request.
+The token is stored as cookie and is presented by the client at each request to LXD.
+
+For instructions on how to set up Candid-based authentication, see the [Candid authentication for LXD](https://ubuntu.com/tutorials/candid-authentication-lxd) tutorial.
+
+(authentication-rbac)=
+## Role Based Access Control (RBAC)
+
+LXD supports integrating with the Canonical RBAC service.
+Combined with Candid-based authentication, {abbr}`RBAC (Role Based Access Control)` can be used to limit what an API client is allowed to do on LXD.
+
+In such a setup, authentication happens through Candid, while the RBAC service maintains roles to user/group relationships.
+Roles can be assigned to individual projects, to all projects or to the entire LXD instance.
+
+The meaning of the roles when applied to a project is as follows:
+
+- auditor: Read-only access to the project
+- user: Ability to do normal life cycle actions (start, stop, ...),
+        execute commands in the instances, attach to console, manage snapshots, ...
+- operator: All of the above + the ability to create, re-configure and
+            delete instances and images
+- admin: All of the above + the ability to reconfigure the project itself
+
+```{important}
+Currently, only the `auditor` and the `user` roles are suitable for users that you wouldn't trust with root access to the host.
+```
```

Comment:
I a follow-up change, we should probably rephrase that as it's no longer quite true.
The statement remains valid unless the project is restricted, restricted projects can be made to be safe for the `operator` role too. `admin` remains dangerous as it allows reconfiguring the project and therefore turning off restrictions.

Reply:
Updated. Can we clarify what we mean with "restricted project"? Is this the same kind of restriction as mentioned in line 46 (https://github.com/lxc/lxd/pull/9787/files#diff-a781e12280f919cbdeeb19978c38c25508af7dbc1decd3f00d94863ec27bcdffR46)?

Reply:
Yeah, it refers to https://linuxcontainers.org/lxd/docs/master/projects/#project-restrictions

Such projects don't have access (unless specifically configured by the administrator) to any of those LXD features that could allow taking over the whole system.

Reply:
Great, I added a link.